engine-src/Game/LambdaHack/Client/UI/TutorialHints.hs

```diff
@@ -0,0 +1,51 @@
+module Game.LambdaHack.Client.UI.TutorialHints
+  (TutorialHints (..)
+  , damageOfDifferentKind
+  , newFloorNewOpportunity
+  , cannotHarmYouInMelee
+  , caughtProjectile
+  , hitsWithNoDirectDamage
+  ) where
+
+import Data.Text (Text)
+import Prelude
+
+data TutorialHints =
+    DamageOfDifferentKind
+    | NewFloorNewOpportunity
+    | CannotHarmYouInMelee
+    | CaughtProjectile
+    | HitsWithNoDirectDamage
+    deriving(Enum, Eq, Show)
+
+
+
+-- | Generate the standard textual representation according to
+--
+renderTutorialHints :: TutorialHints -> Text
+renderTutorialHints DamageOfDifferentKind =
+    "You took damage of a different kind than the normal piercing hit, which means your armor couldn't block any part of it. Normally, your HP (hit points, health) do not regenerate, so losing them is a big deal. Apply healing concoctions or take a long sleep to replenish your HP (but in this hectic environment not even uninterrupted resting that leads to sleep is easy)."
+renderTutorialHints NewFloorNewOpportunity =
+    "New floor is new opportunities, though the old level is still there and others may roam it after you left. Viewing all floors, without moving between them, can be done using the '<' and '>' keys."
+renderTutorialHints CannotHarmYouInMelee =
+    "This enemy can't harm you in melee. Left alone could it possibly be of some use?"
+renderTutorialHints CaughtProjectile =
+    "You managed to catch a projectile, thanks to being braced and hitting it exactly when it was at arm's reach. The obtained item has been put into the shared stash of your party."
+renderTutorialHints HitsWithNoDirectDamage =
+    "Some hits don't cause piercing, impact, burning nor any other direct damage. However, they can have other effects, bad, good or both."
+
+
+damageOfDifferentKind :: Text
```

Comment:
That's too much work. I'd rather a single function that takes a value `v` of type `TutorialHints` and does the same as `msgAdd MsgTutorialHint v`.